### PR TITLE
Add unrestricted keyword argument to uuidToObject()

### PR DESCRIPTION
**Pull request by anirudhhkashyap**
Fixes: [#1448](https://github.com/plone/plone.restapi/issues/1448)

Changes:
- Pass unrestricted keyword argument to uuidToObject()
- The unrestricted argument value is set to True

Setting the unrestricted flag to True invokes the unrestrictedTraversal() method instead of the restrictedTraversal() method as shown [here](https://github.com/plone/plone.app.uuid/blob/65cdc16a81a9ed12453829fd3fd37d3aca5ac328/plone/app/uuid/utils.py#L94-L96). Earlier, invoking the latter threw the Unauthorized Exception. Now the access to the object can be controlled using the unrestricted flag. This will help in deprecating uuidToObject() from plone.outputfilters.browser.resolveuid.

### CODE REVIEW

**Comment by anirudhhkashyap** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/serializer/utils.py [hunk 1]
```diff
@@ -23,7 +23,10 @@ def uid_to_url(path):
     if suffix:
         href += "/" + suffix
     else:
-        target_object = uuidToObject(uid)
+        """ Pass unrestricted flag as true so the object is accessible.
+        At uuidToObject(), this leads to unrestrictedTraverse() to be invoked instead of restrictedTraverse().
+        """
+        target_object = uuidToObject(uid, unrestricted = True)
```
Comment (jensens) [inline src/plone/restapi/serializer/utils.py:29]:
```suggestion
        target_object = uuidToObject(uid, unrestricted=True)
```

**Comment by jensens** [comment]
ad `unrestricted`: Question is, why would we want to get objects we are not allowed to view? That sounds odd.

**Comment by davisagli** [comment]
@jensens Maybe a user has permission to edit an item that includes references to a different item that they do not have permission to view. But, I agree that it is not obvious whether this is the right design for every use case, and might need a more careful analysis of everywhere this function is used.

**Comment by jensens** [comment]
If the user has no `View` permission for the target, we need to check if the user has at least `Access content information`. If both are missing, we can verify the UUID (does target exist) and keep the UUID. But we are not able to display anything else than the plain UUID.

**Comment by jensens** [comment]
I know took a deeper look here. The main problem is, the purpose of the existing code this PR changes is not very clear at the first look. The purpose is to fix the test here:

https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/tests/test_resolveuid.py#L221-L232

It check if an anonymous user can resolve a UUID to a non-published page. This is probably fine, it takes the resolveUID url and return an URL to the actual object. 

The test fails if the helper function touched in this PR is not modified. IMO the test is fine. Having an URL rendered is fine as long as when clicking on the link gives an unauthorized.

I propose - while at it - to add a docstring to the function explaining its purpose, fix the minor code style glitch and the merge the PR.

File: src/plone/restapi/serializer/utils.py [hunk 2]
```diff
@@ -23,7 +23,10 @@ def uid_to_url(path):
     if suffix:
         href += "/" + suffix
     else:
-        target_object = uuidToObject(uid)
+        """ Pass unrestricted flag as true so the object is accessible.
+        At uuidToObject(), this leads to unrestrictedTraverse() to be invoked instead of restrictedTraverse().
+        """
```
Comment (jensens) [inline src/plone/restapi/serializer/utils.py:28]:
```suggestion
        # Pass unrestricted flag as true so the object is accessible.
        # At uuidToObject(), this leads to unrestrictedTraverse() to be invoked instead of restrictedTraverse().
```

**Review by anirudhhkashyap — COMMENTED** [review]
@jensens Sorry for the lack of clarity. I'm kinda new to this project. Thanks for the feedback!

**Review by jensens — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
This needs your PR https://github.com/plone/plone.app.uuid/pull/14 which was merged, but is not in a release yet. And it won't be in Plone 5.2, with `plone.restapi` is still being tested against. So I cannot imagine that this works. Maybe this corner case is not caught by tests.

I would propose to leave `plone.restapi` unchanged for now. Instead fix https://github.com/plone/plone.outputfilters/issues/52
Once we drop 5.2 support in `plone.restapi`, we can revisit this.

**Comment by wesleybl** [comment]
@anirudhhkashyap I reverted this PR at #1475, because it depends on a `plone.app.uuid` release. See:

https://github.com/plone/plone.restapi/pull/1471#issuecomment-1225594991

Can you please open another PR with these changes here? Then we can leave it waiting for a new release of `plone.app.uuid`.

**Comment by anirudhhkashyap** [comment]
> @anirudhhkashyap I reverted this PR at #1475, because it depends on a `plone.app.uuid` release. See:
> 
> [#1471 (comment)](https://github.com/plone/plone.restapi/pull/1471#issuecomment-1225594991)
> 
> Can you please open another PR with these changes here? Then we can leave it waiting for a new release of `plone.app.uuid`.

Sure, no problem.

**Comment by wesleybl** [comment]
@anirudhhkashyap and please, create a branch in plone.restapi instead of make in a fork.
